packages/typescriptlang-org/src/templates/pages/dt/search.tsx

```diff
@@ -0,0 +1,65 @@
+import { graphql } from "gatsby"
+import React, { useEffect, useState } from "react"
+import { debounce } from "ts-debounce"
+import { Layout } from "../../../components/layout"
+import { Intl } from "../../../components/Intl"
+
+import "./search.scss"
+
+import { DTSearchPageQuery } from "../../../__generated__/gatsby-types"
+import { SearchArea } from "../../../components/search/SearchArea"
+import { SearchResultsDisplay } from "../../../components/search/SearchResultsDisplay"
+import { useSearchResult } from "../../../components/search/useSearchResult"
+
+type SearchProps = {
+  data: DTSearchPageQuery
+  location: Location
+  pageContext: any
+}
+
+const updateHistorySearch = debounce((search: string) => {
+  const params = new URLSearchParams(window.location.search)
+  params.set("search", search)
+  history.pushState(null, "", `${window.location.pathname}?${params}`)
+}, 250)
```

Comment:
This was a [fun little reminder](https://twitter.com/JoshuaKGoldberg/status/1302453905813536768) on how to properly investigate performance issues 😄. Calling this function repeatedly was slowing the page's render down.